### ISSUE TITLE
compile error; fn returning fn

### ISSUE THREAD

**Issue by mvlootman** [vlang/v#4687]
**V version: V 0.1.26 50a8373.8df7637** 
**OS: macOS ** 

**What did you do?**
```
// foo returns function fn int->int
fn foo() (fn(int) int ) {
	double_fn := fn(n int) int {
		return  n + n
    }
	return double_fn
}

fn main() {
	df := foo()

	x := df(3)
	println(x)	// expect 6

}
```

**What did you expect to see?**
Able to execute the function.
 
**What did you see instead?**
```
func.v:6:9: error: cannot use `anon_115_7_7` as type `int` in return argument 
    4 |         return  n + n
    5 |     }
    6 |     return double_fn
      |            ~~~~~~~~~
    7 | }
    8 |
```

**Comment by nedpals** [comment]
@mvlootman Try creating an alias for that function type and use it as the return type for it

**Comment by mvlootman** [comment]
@nedpals 
I tried this without success:

```
type bar fn(int)int

fn foo() bar { 
	double_fn := bar {
		return  n + n
        }
	return double_fn
}

fn main() {
	df := foo()

	x := df(3)
	println(x)	// expect 6
}
```

gives me

```
4687_func.v:6:3: error: expr(): bad token `return` 
    4 | fn foo() bar { 
    5 |     double_fn := bar {
    6 |         return  n + n
      |         ~~~~~~
    7 |     }
    8 |     return double_fn
```

**Comment by dkull** [comment]
I am looking into it. Assuming it has broken and is relatively straightforward to fix.

**Comment by mvlootman** [comment]
Fixed by #4729
